sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -196,17 +219,68 @@ def _get_metrics_from_dataflow(self):
 
     job_metrics = self._dataflow_client.get_job_metrics(job_id)
     # If the job has terminated, metrics will not change and we can cache them.
-    if self.job_result.is_in_terminal_state():
+    if self.job_result and self.job_result.is_in_terminal_state():
```

Comment:
No caching of metrics when the job result is not provided right? Maybe worth adding in a comment (or maybe it's obvious. I levae it to you).

Reply:
updated comment